### PR TITLE
Document NodeIterator FILTER_REJECT vs FILTER_SKIP

### LINKED ISSUE #23728: Missing NodeFilter.FILTER_SKIP description
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Document/createNodeIterator

### What specific section or headline is this issue about?

`filter` section

### What information was incorrect, unhelpful, or incomplete?

Missing NodeFilter.FILTER_SKIP description.

### What did you expect to see?

I expected to see an explanation of `NodeFilter.FILTER_SKIP`, including why one would choose to use `NodeFilter.FILTER_SKIP` instead of `NodeFilter.FILTER_REJECT`.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/document/createnodeiterator`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Document/createNodeIterator
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/document/createnodeiterator/index.md
* Last commit: https://github.com/mdn/content/commit/da88b2f3a23b9d93f083003f13c06f9d96073f6a
* Document last modified: 2022-09-13T05:21:27.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
Fix https://github.com/mdn/content/issues/23728

### CODE REVIEW

File: files/en-us/web/api/document/createnodeiterator/index.md [hunk 1]
```diff
@@ -49,8 +49,11 @@ createNodeIterator(root, whatToShow, filter)
...
 
 - `filter` {{optional_inline}}
+
   - : A callback function or an object with an `acceptNode()` method. The function or method will be called for each node in the subtree based at root which is accepted as included by the whatToShow flag to determine whether or not to include it in the list of iterable nodes. The method should return one of `NodeFilter.FILTER_ACCEPT`, `NodeFilter.FILTER_REJECT`, or `NodeFilter.FILTER_SKIP`. See the [Example](#examples).
 
+    For `createNodeIterator`, it is equivalent to return `NodeFilter.FILTER_REJECT`, `NodeFilter.FILTER_SKIP`, or any unrecognized value. This node will not be included in the list of iterable nodes, but its children will continue to be iterated over.
```
Comment (wbamberg) [inline files/en-us/web/api/document/createnodeiterator/index.md:55]: For me at least it is easier reading to say:

```suggestion
    For `createNodeIterator`, the values `NodeFilter.FILTER_REJECT` and `NodeFilter.FILTER_SKIP` are equivalent. This node will not be included in the list of iterable nodes, but its children will continue to be iterated over.
```

I think it's reasonable to expect the context of "for createNodeIterator" to be implicit, and  I'm not sure it helps to include "any unrecognized value".

Comment (Josh-Cena) [reply files/en-us/web/api/document/createnodeiterator/index.md:55]: I'm fine to drop the "any value" part. Although I think it helps to be exhaustive, from my research Chrome and Firefox have diverging behavior on returning unrecognized values in `createTreeWalker` filter function, so I didn't update that page as I planned to. However I think "for `createNodeIterator` is helpful to remind readers it's not true for `createTreeWalker`.

Comment (wbamberg) [reply files/en-us/web/api/document/createnodeiterator/index.md]: > However I think "for createNodeIterator is helpful to remind readers it's not true for createTreeWalker.

Yes, I changed my mind about that :).

**Review by wbamberg — APPROVED** [review]
👍

**Pull request merged**